### ISSUE TITLE
Derivation of nested case classes

### ISSUE THREAD

**Issue by anakos** [ccadllc/cedi-config#2]
Hi-

At present, this library does not seem to support full derivation of nested case classes: i.e. - given the following set of case classes

```
final case class Both(left: Wibble, right: That)
final case class Wibble(value: Int)
final case class Wobble(value: List[String])
```

then simply calling ```derived[Both]``` without explicitly defining implicit values for both ```derived[Wibble]``` and ```derived[Wobble]```, will yield compilation errors indicating that a ConfigParser needs to be in scope for each component type x of Both.

Would you welcome a PR supporting the ability to fully auto-derive nested case classes?

Thanks
Alex

**Comment by mpilquist** [comment]
Hi Alex,

Yeah that would great! Note that before merging the PR, we need a CLA signed. There's more info in [CONTRIBUTING](
https://github.com/ccadllc/cedi-config/blob/master/CONTRIBUTING).

Happy New Year

**Comment by anakos** [comment]
Hi Michael-

Great, thanks. I have sent you a PM requesting the CLA - please let me know if you haven't received.

Thanks and happy new year to you as well!
Alex

**Comment by mpilquist** [comment]
Fixed by #3. Thanks!
